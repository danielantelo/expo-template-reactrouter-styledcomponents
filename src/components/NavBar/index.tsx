import React, { FunctionComponent } from 'react';
import { Button, Footer, FooterTab, Icon, Text } from 'native-base';
import { useRoutes } from '../../navigation/routes';

const NavBar: FunctionComponent = () => {
  const { navigateToHome, navigateToSettings } = useRoutes();
  return (
    <Footer>
      <FooterTab>
        <Button vertical onPress={navigateToHome}>
          <Icon name="home" />
          <Text>Apps</Text>
        </Button>
        <Button vertical onPress={navigateToSettings}>
          <Icon name="settings" />
          <Text>Settings</Text>
        </Button>
      </FooterTab>
    </Footer>
  );
};

export default NavBar;
