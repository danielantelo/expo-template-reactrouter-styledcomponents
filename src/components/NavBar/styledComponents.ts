import { View } from 'react-native';
import { StyledComponent } from '../../utils';

export const NavBarContainer = StyledComponent(() => {
  return {
    display: 'flex',
    flexDirection: 'row',
    '@media(min-width: 768px)': {
      position: 'fixed',
      top: 0,
      right: 0,
      background: 'white',
      flexDirection: 'column',
    },
  };
}, View);
