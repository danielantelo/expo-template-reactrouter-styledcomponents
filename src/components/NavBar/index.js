import React from 'react';
import { Text } from 'react-native';
import { HomeLink, SettingsLink } from '../../navigation/routes';
import { NavBarContainer, NavBarItem } from './styledComponents';

const NavBar = ({ float = true }) => {
  return (
    <NavBarContainer nativeID="test" float={float}>
      <NavBarItem>
        <HomeLink><Text>Home</Text></HomeLink>
      </NavBarItem>
      <NavBarItem>
        <SettingsLink><Text>Settings</Text></SettingsLink>
      </NavBarItem>
    </NavBarContainer>
  )
}

export default NavBar;
