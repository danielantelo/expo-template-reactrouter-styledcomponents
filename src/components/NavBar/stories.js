import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { MemoryRouter } from 'react-router';
import NavBar from './';

storiesOf('Components/NavBar', module)
  .addDecorator((story) => <MemoryRouter>{story()}</MemoryRouter>)
  .add('default', () => <NavBar />);
