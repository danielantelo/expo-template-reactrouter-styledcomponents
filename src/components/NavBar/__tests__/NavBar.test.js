import React from 'react';
import { create } from 'react-test-renderer';
import NavBar from '..';

describe('NavBar', () => {
  const renderer = create(<NavBar />);

  it('renders', () => {
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
