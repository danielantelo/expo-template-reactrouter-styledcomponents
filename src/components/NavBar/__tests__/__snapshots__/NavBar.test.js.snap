// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders 1`] = `
<View
  className="sc-AykKC fQNdfl"
>
  <View
    className="sc-AykKD jHLvSh"
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <Text
        style={null}
      >
        Home
      </Text>
    </View>
  </View>
  <View
    className="sc-AykKD jHLvSh"
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <Text
        style={null}
      >
        Settings
      </Text>
    </View>
  </View>
</View>
`;
